packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_details.dart

```diff
@@ -4,53 +4,202 @@
 import '../controller/model.dart';
-import 'class_stats_table.dart';
 
 class HeapClassDetails extends StatelessWidget {
   const HeapClassDetails({
     Key? key,
+    required this.controller,
     required this.item,
-    required this.sorting,
   }) : super(key: key);
 
-  final SnapshotListItem item;
-  final ColumnSorting sorting;
+  final DiffPaneController controller;
+  final SnapshotInstanceItem item;
 
   @override
   Widget build(BuildContext context) {
-    return ValueListenableBuilder<HeapClassStatistics?>(
+    return ValueListenableBuilder<ClassStats?>(
       valueListenable: item.selectedClassStats,
-      builder: (_, record, __) {
-        if (record == null) {
+      builder: (_, classStats, __) {
+        if (classStats == null) {
           return const Center(
             child: Text('Select class to see details here.'),
           );
         }
-        if (item.diffWith.value == null) {
-          return ClassStatsTable(
-            data: record,
-            sorting: sorting,
-          );
-        }
 
-        return _DiffClassDetails(item: item);
+        return _RetainingPathTable(
+          data: classStats,
+          sorting: controller.classSorting,
+          controller: controller,
+          item: item,
+        );
       },
     );
   }
 }
 
-class _DiffClassDetails extends StatelessWidget {
-  const _DiffClassDetails({Key? key, required this.item}) : super(key: key);
-  final SnapshotListItem item;
+class _RetainingPathColumn extends ColumnData<StatsByPathEntry> {
+  _RetainingPathColumn()
+      : super.wide(
+          'Retaining Path',
+          titleTooltip: 'Class names of objects that retain'
+              '\nthe instances from garbage collection.',
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  String? getValue(StatsByPathEntry record) => record.key.asShortString();
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  String getTooltip(StatsByPathEntry record) => record.key.asLongString();
+}
+
+class _InstanceColumn extends ColumnData<StatsByPathEntry> {
+  _InstanceColumn()
+      : super(
+          'Instances',
+          titleTooltip: 'Number of instances of the class\n'
+              'retained by the path.',
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(StatsByPathEntry record) => record.value.instanceCount;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class _ShallowSizeColumn extends ColumnData<StatsByPathEntry> {
+  _ShallowSizeColumn()
+      : super(
+          'Shallow\nDart Size',
+          titleTooltip: shallowSizeColumnTooltip,
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(StatsByPathEntry record) => record.value.shallowSize;
+
+  @override
+  bool get supportsSorting => true;
```

Comment:
for numeric columns, this is redundant. On the superclass `supportsSorting => numeric`